### PR TITLE
ChatFilter: Add config section for Filter Lists

### PR DESCRIPTION
**Pull request by Broooklyn**
The chat filter config can get pretty unruly if you have a lot of word and regex filters. This adds a config section akin to the `Item Lists` section in `Ground Items`.  `Filter type` config is shifted down to be with the rest of the configs.

Screen Shot 2020-06-14 at 6 26 23 PM

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterConfig.java [hunk 1]
```diff
@@ -113,7 +125,7 @@ default boolean filterLogin()
 		keyName = "collapseGameChat",
 		name = "Collapse Game Chat",
 		description = "Collapse duplicate game chat messages into a single line",
-		position = 9
+		position = 8
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterConfig.java:128]: Why are these positions changed?

File: runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterConfig.java [hunk 2]
```diff
@@ -28,26 +28,25 @@
...
 {
-	@ConfigItem(
-		keyName = "filterType",
-		name = "Filter type",
-		description = "Configures how the messages are filtered",
-		position = 1
+	@ConfigSection(
+		name = "Filter Lists",
+		description = "Custom Word, Regex, and Username filter lists",
+		position = 0,
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterConfig.java:39]: Just make this position 1 like "Filter type" was before

File: runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterConfig.java (hunk 1 above)
Comment (Broooklyn) [reply runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterConfig.java:128]: iirc my intent was to have the positions reflect their actual order, making it as clear as possible when viewing `ChatFilterConfig`. Previously it went from 7 to 9, so I corrected it.

File: runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterConfig.java (hunk 2 above)
Comment (Broooklyn) [reply runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterConfig.java:39]: I was just following[ the precedent](https://github.com/runelite/runelite/commit/40f17648b4e83d1659d3fde5b7fb1d3ba2218a04#diff-637c5ee9b4bc1dae6fb2a818d2b34443) set by Hydrox, but I'm happy to change it if you feel I should.

File: runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterConfig.java (hunk 1 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterConfig.java:128]: There's no need for positions to be contiguous. Please revert the unneeded position changes to this file.

**Pull request merged**